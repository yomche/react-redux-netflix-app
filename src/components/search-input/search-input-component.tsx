import React, { FC } from 'react';
import {
  StyledSearchSection,
  StyledHeading,
  StyledSearchField,
} from './search-input-styles';

export const SearchInput: FC = () => (
  <StyledSearchSection>
    <StyledHeading>FIND YOUR MOVIE</StyledHeading>
    <StyledSearchField type="text" placeholder="Quentin Tarantino" />
  </StyledSearchSection>
);
