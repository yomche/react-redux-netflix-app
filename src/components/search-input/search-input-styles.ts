import styled from 'styled-components';

const BACKGROUND_COLOR = '#000000';
const FONT_COLOR = '#ffffff';
const BORDER_COLOR = '#ff0000';

export const StyledSearchSection = styled.form`
    width: 1000px;
    margin: auto;
`;

export const StyledHeading = styled.h1`
    color: ${FONT_COLOR};
    font-weight: bold;
    font-size: 35px;
    font-family: "Lucida Sans Unicode", "Lucida Grande", sans-serif;
`;

export const StyledSearchField = styled.input`
    width: 1000px;
    height: 30px;
    font-size: 20px;
    color: ${FONT_COLOR};
    background-color: ${BACKGROUND_COLOR};
    border-bottom: 2px solid ${BORDER_COLOR};
`;
