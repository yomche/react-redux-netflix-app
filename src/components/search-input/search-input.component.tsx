import React, { FC } from 'react';
import { Form } from 'react-final-form';
import {
  StyledSearchForm,
  StyledHeading,
  StyledSearchInput,
  StyledControlsSection,
  StyledTitleButton,
  StyledGenreButton,
  StyledSubmitButton,
} from './search-input.styles';

type SearchInputProps = {
  inputValue: string;
  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
};

export const SearchInput: FC<SearchInputProps> = ({ handleChange, inputValue }) => (
  <Form
    onSubmit={() => {
      // eslint-disable-next-line no-console
      console.log(`Введенный фильм: ${inputValue}`);
    }}
    render={({ handleSubmit }) => (
      <StyledSearchForm onSubmit={handleSubmit}>
        <StyledHeading>FIND YOUR MOVIE</StyledHeading>
        <StyledSearchInput
          type="text"
          value={inputValue}
          placeholder="Quentin Tarantino"
          onChange={handleChange}
        />
        <StyledControlsSection>
          <StyledTitleButton> TITLE </StyledTitleButton>
          <StyledGenreButton> GENRE </StyledGenreButton>
          <StyledSubmitButton> SUBMIT </StyledSubmitButton>
        </StyledControlsSection>
      </StyledSearchForm>
    )}
  />
);
