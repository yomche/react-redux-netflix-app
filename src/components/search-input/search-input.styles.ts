import styled from 'styled-components';

const BACKGROUND_COLOR = '#000000';
const FONT_COLOR = '#ffffff';
const BORDER_COLOR = '#ff0000';
const BUTTON_BACKGROUND_COLOR = '#ff0000';
const BUTTON_COLOR = '#a9a9a9';

export const StyledSearchForm = styled.form`
  width: 1000px;
  margin: auto;
`;

export const StyledInput = styled.div`
  & > input {
    width: 1000px;
    height: 30px;
    font-size: 20px;
    color: ${FONT_COLOR};
    background-color: ${BACKGROUND_COLOR};
    border-bottom: 2px solid ${BORDER_COLOR};
  }
`;

export const StyledError = styled.span`
  color: ${FONT_COLOR};
  font-weight: bold;
`;

export const StyledHeading = styled.h1`
  color: ${FONT_COLOR};
  font-weight: bold;
  font-size: 35px;
`;

export const StyledSearchInput = styled.input`
  width: 1000px;
  height: 30px;
  font-size: 20px;
  color: ${FONT_COLOR};
  background-color: ${BACKGROUND_COLOR};
  border-bottom: 2px solid ${BORDER_COLOR};
`;

export const StyledControlsSection = styled.section`
  width: 1000px;
  height: 70px;
  margin: auto;
  display: flex;
  align-items: center;
`;

export const StyledTitleButton = styled.label`
  width: 100px;
  height: 25px;
  color: ${FONT_COLOR};
  background-color: ${BUTTON_BACKGROUND_COLOR};
  margin-right: 15px;
  border-radius: 5px;
  text-align: center;

  :hover {
    cursor: pointer;
    background-color: ${BUTTON_COLOR};
  }

  & > input[type='radio'] {
    display: none;
  }
`;

export const StyledGenreButton = styled.label`
  width: 100px;
  height: 25px;
  color: ${FONT_COLOR};
  background-color: ${BUTTON_BACKGROUND_COLOR};
  border-radius: 5px;
  margin-right: 650px;
  text-align: center;

  :hover {
    cursor: pointer;
    background-color: ${BUTTON_COLOR};
  }

  & > input[type='radio'] {
    display: none;
  }
`;

export const StyledSubmitButton = styled.button`
  width: 160px;
  height: 35px;
  font-size: 17px;
  color: ${FONT_COLOR};
  background-color: ${BUTTON_BACKGROUND_COLOR};
  margin-top: 20px;
  border-radius: 5px;

  :hover {
    cursor: pointer;
    background-color: ${BUTTON_COLOR};
  }
`;
