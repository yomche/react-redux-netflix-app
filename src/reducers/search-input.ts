import { searchInputValueAction } from '../actions/action-names';

export const initialState = '';

export function searchInputValueData(
  state = initialState,
  action: { type: string; payload: string }
): string {
  switch (action.type) {
    case searchInputValueAction:
      return action.payload;
    default:
      return state;
  }
}
