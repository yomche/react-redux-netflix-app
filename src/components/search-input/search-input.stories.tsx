/* eslint-disable @typescript-eslint/no-empty-function */
import React, { ComponentProps } from 'react';

import { Story } from '@storybook/react';

import { SearchInput } from './search-input.component';

export default {
  title: 'Components/Search Input',
  component: SearchInput,
};

const Template: Story<ComponentProps<typeof SearchInput>> = () => {
  const movieSearchType = 'test value';
  function onSetSearchInputValue() {}
  function onSetMovieSearchType() {}

  return (
    <SearchInput
      updateSearchInputValue={onSetSearchInputValue}
      toggleMovieSearchType={onSetMovieSearchType}
      movieSearchType={movieSearchType}
    />
  );
};

export const Primary = Template.bind({});
Primary.args = {};
