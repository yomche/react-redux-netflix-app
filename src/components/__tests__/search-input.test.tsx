import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { SearchInput } from '../search-input/search-input.component';

describe('< SearchInput /> component', () => {
  const MockInput = {
    movieSearchType: 'title',
    onSetSearchInputValue: jest.fn(),
    onSetMovieSearchType: jest.fn(),
  };
  test('renders correctly', () => {
    const { asFragment } = render(
      <Router>
        <SearchInput
          updateSearchInputValue={MockInput.onSetSearchInputValue}
          setMovieSearchType={MockInput.onSetMovieSearchType}
          movieSearchType={MockInput.movieSearchType}
        />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
