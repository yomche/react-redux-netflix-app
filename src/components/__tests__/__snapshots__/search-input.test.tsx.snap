// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< SearchInput /> component renders correctly 1`] = `
<DocumentFragment>
  <form
    class="StyledSearchForm-sc-1uqbsb-0 bMzDOT"
  >
    <h1
      class="StyledHeading-sc-1uqbsb-3 jjwjis"
    >
      FIND YOUR MOVIE
    </h1>
    <div
      class="StyledInput-sc-1uqbsb-1 jrqfiq"
    >
      <input
        name="inputValue"
        placeholder="Quentin Tarantino"
        type="text"
        value=""
      />
    </div>
    <section
      class="StyledControlsSection-sc-1uqbsb-5 llnDDy"
    >
      <label
        class="StyledTitleButton-sc-1uqbsb-6 ipEjeb"
      >
        <input
          checked=""
          name="searchType"
          type="radio"
          value="title"
        />
         TITLE
      </label>
      <label
        class="StyledGenreButton-sc-1uqbsb-7 XQUYS"
      >
        <input
          name="searchType"
          type="radio"
          value="genres"
        />
         GENRE
      </label>
      <button
        class="StyledSubmitButton-sc-1uqbsb-8 kfslJL"
        type="submit"
      >
         SUBMIT 
      </button>
    </section>
  </form>
</DocumentFragment>
`;
